### PR TITLE
[Feature] display resource group info in explain verbose

### PR DESCRIPTION
**Pull request by murphyatwork**
## What type of PR is this：
- [ ] bug
- [x] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

Use `explain verbose` to display resource group information. And refactor some chaos code BTW.

E.g.

```sql
MySQL root@127.1:hehe> explain verbose select count(*) from t0;
+--------------------------------------------------------------------------------------------------+
| Explain String                                                                                   |
+--------------------------------------------------------------------------------------------------+
| RESOURCE GROUP: rg_mofei                                                                         |
|                                                                                                  |
| PLAN FRAGMENT 0(F00)                                                                             |
|   Output Exprs:3: count                                                                          |
|   Input Partition: RANDOM                                                                        |
|   RESULT SINK                                                                                    |
|                                                                                                  |
|   1:AGGREGATE (update finalize)                                                                  |
|   |  aggregate: count[(*); args: ; result: BIGINT; args nullable: false; result nullable: false] |
|   |  cardinality: 1                                                                              |
|   |                                                                                              |
|   0:OlapScanNode                                                                                 |
|      table: t0, rollup: t0                                                                       |
|      preAggregation: on                                                                          |
|      partitionsRatio=1/1, tabletsRatio=1/1                                                       |
|      tabletList=41030                                                                            |
|      actualRows=1, avgRowSize=1.0                                                                |
|      cardinality: 1                                                                              |
+--------------------------------------------------------------------------------------------------+
```

### CODE REVIEW

**Comment by mofeiatwork** [comment]
@Mergifyio backport branch-2.3 branch-2.2

**Comment by mofeiatwork** [comment]
run starrocks_fe_unittest

**Comment by mofeiatwork** [comment]
run starrocks_fe_unittest

**Comment by mofeiatwork** [comment]
@Mergifyio rebase

**Comment by mofeiatwork** [comment]
@Mergifyio rebase

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 35 / 63 (55.56%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/Coordinator.java|20|44|45.45%|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|15|19|78.95%|

File: fe/fe-core/src/main/java/com/starrocks/qe/ConnectProcessor.java [hunk 1]
```diff
@@ -226,7 +226,6 @@ private void addFinishedQueryDetail() {
         }
         queryDetail.setEndTime(endTime);
         queryDetail.setLatency(elapseMs);
-        queryDetail.setWorkGroupName(ctx.getWorkGroup() != null ? ctx.getWorkGroup().getName() : "");
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/qe/ConnectProcessor.java:229]: Why remove this?  why QueryDetail needn't work group info?

File: fe/fe-core/src/main/java/com/starrocks/qe/Coordinator.java [hunk 2]
```diff
@@ -611,13 +662,10 @@ private void deliverExecFragmentRequests(boolean enablePipelineEngine) throws Ex
...
 
-                    boolean needCheckBackendState = false;
-                    if (queryOptions.getQuery_type() == TQueryType.LOAD && profileFragmentId == 0) {
-                        // this is a load process, and it is the first fragment.
-                        // we should add all BackendExecState of this fragment to needCheckBackendExecStates,
-                        // so that we can check these backends' state when joining this Coordinator
-                        needCheckBackendState = true;
-                    }
+                    boolean needCheckBackendState = queryOptions.getQuery_type() == TQueryType.LOAD && profileFragmentId == 0;
+                    // this is a load process, and it is the first fragment.
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/qe/Coordinator.java:666]: these comments should above 665 line.

File: fe/fe-core/src/main/java/com/starrocks/qe/StmtExecutor.java [hunk 3]
```diff
@@ -703,19 +696,21 @@ private void handleSetStmt() {
...
-                                 List<String> colNames, List<Expr> outputExprs, String explainString) throws Exception {
+    private void handleQueryStmt(ExecPlan execPlan) throws Exception {
         // Every time set no send flag and clean all data in buffer
         context.getMysqlChannel().reset();
-        StatementBase queryStmt = parsedStmt;
 
-        if (queryStmt.isExplain()) {
-            handleExplainStmt(explainString);
+        if (parsedStmt.isExplain()) {
+            handleExplainStmt(execPlan);
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/qe/StmtExecutor.java:704]: when context.getSessionVariable().isReportSucc(), also need to handleExplainStmt

File: fe/fe-core/src/main/java/com/starrocks/qe/ConnectProcessor.java (hunk 1 above)
Comment (mofeiatwork) [reply fe/fe-core/src/main/java/com/starrocks/qe/ConnectProcessor.java:229]: The workgroup field is not used.

File: fe/fe-core/src/main/java/com/starrocks/qe/Coordinator.java (hunk 2 above)
Comment (mofeiatwork) [reply fe/fe-core/src/main/java/com/starrocks/qe/Coordinator.java:666]: done

File: fe/fe-core/src/main/java/com/starrocks/qe/StmtExecutor.java (hunk 3 above)
Comment (mofeiatwork) [reply fe/fe-core/src/main/java/com/starrocks/qe/StmtExecutor.java:704]: I'm afraid not. 

Previous code `parsedStmt.isExplain() || context.getSessionVariable().isReportSucc()` means needs to generate `explainString`, but only `parsedStmt.isExplain()` need to response result as `explain sql`.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 35 / 63 (55.56%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/Coordinator.java|20|44|45.45%|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|15|19|78.95%|

**Review by satanson — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/qe/StmtExecutor.java (hunk 3 above)
Comment (kangkaisen) [reply fe/fe-core/src/main/java/com/starrocks/qe/StmtExecutor.java:704]:
```
        if (context.getQueryDetail() != null) {
            context.getQueryDetail().setExplain(explainString);
        }
```
you should keep explain string in QueryDetail, workGroup also need to be kept in QueryDetail

Comment (mofeiatwork) [reply fe/fe-core/src/main/java/com/starrocks/qe/StmtExecutor.java:704]: fixed

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

File: fe/fe-core/src/main/java/com/starrocks/qe/StmtExecutor.java [hunk 4]
```diff
@@ -703,19 +696,21 @@ private void handleSetStmt() {
...
-    private void handleQueryStmt(List<PlanFragment> fragments, List<ScanNode> scanNodes, TDescriptorTable descTable,
-                                 List<String> colNames, List<Expr> outputExprs, String explainString) throws Exception {
+    private void handleQueryStmt(ExecPlan execPlan) throws Exception {
         // Every time set no send flag and clean all data in buffer
         context.getMysqlChannel().reset();
-        StatementBase queryStmt = parsedStmt;
 
-        if (queryStmt.isExplain()) {
-            handleExplainStmt(explainString);
+        if (parsedStmt.isExplain()) {
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/qe/StmtExecutor.java:703]: This condition seems also need to change?

**Comment by mofeiatwork** [comment]
run starrocks_fe_unittest

**Comment by mofeiatwork** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 37 / 67 (55.22%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/Coordinator.java|20|44|45.45%|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|17|23|73.91%|

**Comment by mofeiatwork** [comment]
@Mergifyio rebase

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 37 / 67 (55.22%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/Coordinator.java|20|44|45.45%|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|17|23|73.91%|

**Review by kangkaisen — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/qe/Coordinator.java [hunk 5]
```diff
@@ -693,19 +741,19 @@ private void deliverExecFragmentRequests(boolean enablePipelineEngine) throws Ex
...
      * │    │  │    │
      * 5    6  │    │
-     *      │  │    │
-     *      └──┼────┘
-     *         │
-     *         7
+     * │  │    │
+     * └──┼────┘
+     * │
+     * 7
```
Comment (ZiheLiu) [inline fe/fe-core/src/main/java/com/starrocks/qe/Coordinator.java:756]: Formatting this code destroys the ASCII graph.

Comment (mofeiatwork) [reply fe/fe-core/src/main/java/com/starrocks/qe/Coordinator.java]: done

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by mofeiatwork** [comment]
@Mergifyio rebase

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 37 / 67 (55.22%)




#### file detail

|   |path|covered line|new line|coverage|not covered line detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/Coordinator.java|20|44|45.45%|[459, 460, 461, 463, 464, 465, 467, 468, 500, 503, 504, 505, 509, 510, 511, 512, 516, 517, 518, 522, 523, 524, 527, 2724]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|17|23|73.91%|[698, 702, 940, 963, 964, 965]|

File: fe/fe-core/src/main/java/com/starrocks/qe/StmtExecutor.java [hunk 6]
```diff
@@ -1066,12 +1083,9 @@ public void handleInsertOverwrite(InsertStmt insertStmt) {
 
     public void handleDMLStmt(ExecPlan execPlan, DmlStmt stmt) throws Exception {
         if (stmt.isExplain()) {
-            handleExplainStmt(execPlan.getExplainString(stmt.getExplainLevel()));
+            handleExplainStmt(buildExplainString(execPlan));
             return;
         }
-        if (context.getQueryDetail() != null) {
-            context.getQueryDetail().setExplain(execPlan.getExplainString(TExplainLevel.NORMAL));
-        }
```
Comment (ZiheLiu) [inline fe/fe-core/src/main/java/com/starrocks/qe/StmtExecutor.java:1074]: Is it possible that `stmt.isExplain==false` and `context.getQueryDetail() != null`?

Comment (mofeiatwork) [reply fe/fe-core/src/main/java/com/starrocks/qe/StmtExecutor.java:1074]: fixed

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by ZiheLiu — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 36 / 67 (53.73%)




#### file detail

|   |path|covered line|new line|coverage|not covered line detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/Coordinator.java|20|44|45.45%|[459, 460, 461, 463, 464, 465, 467, 468, 500, 503, 504, 505, 509, 510, 511, 512, 516, 517, 518, 522, 523, 524, 527, 2724]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|16|23|69.57%|[698, 702, 940, 962, 963, 964, 1088]|

**Review by kangkaisen — APPROVED** [review]

**Review by satanson — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 36 / 67 (53.73%)




#### file detail

|   |path|covered line|new line|coverage|not covered line detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/Coordinator.java|20|44|45.45%|[459, 460, 461, 463, 464, 465, 467, 468, 500, 503, 504, 505, 509, 510, 511, 512, 516, 517, 518, 522, 523, 524, 527, 2724]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|16|23|69.57%|[698, 702, 940, 962, 963, 964, 1088]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 36 / 67 (53.73%)




#### file detail

|   |path|covered line|new line|coverage|not covered line detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/Coordinator.java|20|44|45.45%|[459, 460, 461, 463, 464, 465, 467, 468, 500, 503, 504, 505, 509, 510, 511, 512, 516, 517, 518, 522, 523, 524, 527, 2724]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|16|23|69.57%|[698, 702, 940, 962, 963, 964, 1088]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 36 / 67 (53.73%)




#### file detail

|   |path|covered line|new line|coverage|not covered line detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/Coordinator.java|20|44|45.45%|[459, 460, 461, 463, 464, 465, 467, 468, 500, 503, 504, 505, 509, 510, 511, 512, 516, 517, 518, 522, 523, 524, 527, 2724]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|16|23|69.57%|[698, 702, 940, 962, 963, 964, 1088]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 36 / 67 (53.73%)




#### file detail

|   |path|covered line|new line|coverage|not covered line detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/Coordinator.java|20|44|45.45%|[459, 460, 461, 463, 464, 465, 467, 468, 500, 503, 504, 505, 509, 510, 511, 512, 516, 517, 518, 522, 523, 524, 527, 2724]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|16|23|69.57%|[698, 702, 940, 962, 963, 964, 1088]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 36 / 67 (53.73%)




#### file detail

|   |path|covered line|new line|coverage|not covered line detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/Coordinator.java|20|44|45.45%|[459, 460, 461, 463, 464, 465, 467, 468, 500, 503, 504, 505, 509, 510, 511, 512, 516, 517, 518, 522, 523, 524, 527, 2724]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|16|23|69.57%|[698, 702, 940, 962, 963, 964, 1088]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 36 / 67 (53.73%)




#### file detail

|   |path|covered line|new line|coverage|not covered line detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/Coordinator.java|20|44|45.45%|[459, 460, 461, 463, 464, 465, 467, 468, 500, 503, 504, 505, 509, 510, 511, 512, 516, 517, 518, 522, 523, 524, 527, 2724]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|16|23|69.57%|[698, 702, 940, 962, 963, 964, 1088]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 36 / 67 (53.73%)




#### file detail

|   |path|covered line|new line|coverage|not covered line detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/Coordinator.java|20|44|45.45%|[459, 460, 461, 463, 464, 465, 467, 468, 500, 503, 504, 505, 509, 510, 511, 512, 516, 517, 518, 522, 523, 524, 527, 2724]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|16|23|69.57%|[698, 702, 940, 962, 963, 964, 1088]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 36 / 67 (53.73%)




#### file detail

|   |path|covered line|new line|coverage|not covered line detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/Coordinator.java|20|44|45.45%|[459, 460, 461, 463, 464, 465, 467, 468, 500, 503, 504, 505, 509, 510, 511, 512, 516, 517, 518, 522, 523, 524, 527, 2724]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|16|23|69.57%|[698, 702, 940, 962, 963, 964, 1088]|

**Pull request merged**